File: specs/NativeMyModule.ts
import { TurboModule, TurboModuleRegistry } from 'react-native';

export interface Spec extends TurboModule {
  readonly getDeviceName: () => string;
  readonly multiply: (a: number, b: number) => Promise<number>;
}

export default TurboModuleRegistry.getEnforcing<Spec>('MyModule');

File: android/app/src/main/java/com/rnbaseproj/NativeMyModule.kt
package com.rnbaseproj

import com.facebook.fbreact.specs.NativeMyModuleSpec
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.Promise

class MyModule(reactContext: ReactApplicationContext) : NativeMyModuleSpec(reactContext) {

    override fun getName(): String = NAME

    override fun getDeviceName(): String {
        return android.os.Build.MODEL
    }

    override fun multiply(a: Double, b: Double, promise: Promise) {
        promise.resolve(a * b)
    }

    companion object {
        const val NAME = "MyModule"
    }
}